backends/s3/s3_test.go

```diff
@@ -74,24 +78,27 @@ func getBackend(ctx context.Context, t *testing.T) (b *Backend) {
 
 func TestBackend(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	t.Cleanup(cancel)
+	defer cancel()
 
 	b := getBackend(ctx, t)
 	tester.DoBackendTests(t, b)
-	assert.Equal(t, "", b.lastMarker)
+	assert.Len(t, b.lastMarker, 0)
 }
 
 func TestBackend_marker(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	t.Cleanup(cancel)
+	defer cancel()
 
 	b := getBackend(ctx, t)
 	b.opt.UseUpdateMarker = true
 
 	tester.DoBackendTests(t, b)
-	assert.Equal(t, "bar-1", b.lastMarker)
 
-	data, err := b.Load(ctx, UpdateMarkerFilename)
+	markerFileContent, err := b.Load(ctx, UpdateMarkerFilename)
+	assert.NoError(t, err)
+	assert.True(t, bytes.HasPrefix(markerFileContent, []byte("foo-1")))
+
+	markerFileContent, err = b.Load(ctx, UpdateMarkerFilename)
 	assert.NoError(t, err)
-	assert.Equal(t, "bar-1", string(data))
+	assert.Equal(t, b.lastMarker, markerFileContent)
```

Comment:
By removing the tests for expected content, this test no longer tests the marker. it could be an empty string here and the test would be totally happy.

Since this test is here to test the marker, test for the specific contents you expect. You cannot control the timestamp and external ETag, but you can check if it starts with the filename.

Reply:
Ah wait, you are testing the prefix above. Then this is fine, but you still need to add back the "bar-1" test.

Reply:
I changed the prefix test in favour of a regexp test.

Not sure if you mean "bar-1", the value, or if you mean the test. In case you mean the value "bar-1" needs to appear, it is no longer true because the last write operation on the backend (in DoBackendTests) is the deletion of the blob named "foo-1".

Reply:
There was one test above that you removed.